Sources/MongoSwift/SDAM.swift

```diff
@@ -57,137 +65,105 @@ public struct ServerDescription {
+        self.minWireVersion = isMaster["minWireVersion"]?.asInt() ?? 0
+        self.maxWireVersion = isMaster["maxWireVersion"]?.asInt() ?? 0
+        self.me = try? isMaster["me"]?.stringValue.map(Address.init)
         self.setName = isMaster["setName"]?.stringValue
-        self.setVersion = isMaster["setVersion"]?.int64Value
+        self.setVersion = isMaster["setVersion"]?.asInt()
         self.electionId = isMaster["electionId"]?.objectIdValue
+        self.primary = try? isMaster["primary"]?.stringValue.map(Address.init)
+        self.logicalSessionTimeoutMinutes = isMaster["logicalSessionTimeoutMinutes"]?.asInt()
 
-        if let primary = isMaster["primary"]?.stringValue {
-            self.primary = ConnectionId(primary)
-        }
+        self.hosts = isMaster["hosts"]?.arrayValue?.asArrayOf(String.self)?.compactMap { host in
+            try? Address(host)
+        } ?? []
 
-        self.logicalSessionTimeoutMinutes = isMaster["logicalSessionTimeoutMinutes"]?.asInt64()
-    }
+        self.passives = isMaster["passives"]?.arrayValue?.asArrayOf(String.self)?.compactMap { passive in
+            try? Address(passive)
+        } ?? []
 
-    /// An internal initializer to create a `ServerDescription` from an OpaquePointer to a
-    /// mongoc_server_description_t.
-    internal init(_ description: OpaquePointer) {
-        self.connectionId = ConnectionId(mongoc_server_description_host(description))
-        self.roundTripTime = mongoc_server_description_round_trip_time(description)
-        // we have to copy because libmongoc owns the pointer.
-        let isMaster = Document(copying: mongoc_server_description_ismaster(description))
-        self.parseIsMaster(isMaster)
+        self.arbiters = isMaster["arbiters"]?.arrayValue?.asArrayOf(String.self)?.compactMap { arbiter in
+            try? Address(arbiter)
+        } ?? []
 
-        let serverType = String(cString: mongoc_server_description_type(description))
-        // swiftlint:disable:next force_unwrapping
-        self.type = ServerType(rawValue: serverType)! // libmongoc will always give us a valid raw value.
+        self.tags = isMaster["tags"]?.documentValue.map { tagsDoc in
+            var tags: [String: String] = [:]
+            for (k, v) in tagsDoc {
+                tags[k] = v.stringValue
+            }
+            return tags
+        } ?? [:]
     }
 }
 
 extension ServerDescription: Equatable {
     public static func == (lhs: ServerDescription, rhs: ServerDescription) -> Bool {
-        // Compare everything except `error` it is always nil and `MongoError`s are not `Equatable`
-        return lhs.connectionId == rhs.connectionId &&
-            lhs.roundTripTime == rhs.roundTripTime &&
-            lhs.lastWriteDate == rhs.lastWriteDate &&
-            lhs.opTime == rhs.opTime &&
+        // As per the SDAM spec, only some fields are necessary to compare for equality.
+        return lhs.address == rhs.address &&
```

Comment:
This will at some point need to account for a stored `error` on the `ServerDescription` as well.